src/tools/rbd/action/Kernel.cc

```diff
@@ -153,6 +153,39 @@ static int do_kernel_showmapped(Formatter *f)
   return r;
 }
 
+static void rbd_check_unsup_features(const char *poolname, const char *imgname,
+				     const char *snapname, int maperrno)
+{
+  int r;
+  uint8_t oldformat;
+  librados::Rados rados;
+  librados::IoCtx ioctx;
+  librbd::Image image;
+
+  /*
+   * kernel returns ENXIO when mapping a V2 image due to unsupported feature
+   * set or due to an invalid header for a V1 image format.
+   */
+  if (maperrno != ENXIO)
```

Comment:
Hmmm.. In that case we need to handle EINVAL too. It might be a good option to handle all error codes except ENOENT and point the user to dmesg.